src/map/entities/battleentity.cpp

```diff
@@ -1950,7 +1950,7 @@ bool CBattleEntity::OnAttack(CAttackState& state, action_t& action)
                     actionTarget.param        = 0;
                     actionTarget.messageID    = 0;
                     actionTarget.spikesEffect = SUBEFFECT_COUNTER;
-                    if (battleutils::IsAbsorbByShadow(this))
+                    if (battleutils::IsAbsorbByShadow(this, PTarget))
```

Comment:
Why did we swap this one? this would still be the attacker and PTarget the defender.

Reply:
Guess it was coded wrong to begin with, I just followed the preexisting setup. I can flip them around.